Custom Constraints

AppDaemon 3.x allows for the creation of custom constraints:

```
self.register_constraint('constrain_cloudy')
```

However, Appdaemon-Test-Framework doesn't seem to understand them:

```
self = <settings.apps.hass.TestApp object at 0x10d11c780>, name = 'constrain_cloudy'

    def register_constraint(self, name):
>       self.constraints.append(name)
E       AttributeError: 'AutoVacationMode' object has no attribute 'constraints'
```

I assume that this needs to be patched somehow, but I'm clueless on this one. 😆 Any thoughts?

**EDIT:** Actually, I think it might be pretty straightforward; we'll see. Submitted [a PR](https://github.com/FlorianKempenich/Appdaemon-Test-Framework/pull/3); appreciate your review!

--- a/appdaemontestframework/init_framework.py
+++ b/appdaemontestframework/init_framework.py
@@ -32,6 +32,8 @@ def patch_hass():
     patch_call_service = mock.patch.object(Hass, 'call_service')
     patch_turn_on = mock.patch.object(Hass, 'turn_on')
     patch_turn_off = mock.patch.object(Hass, 'turn_off')
+    # Custom callback functions
+    patch_register_constraint = mock.patch.object(Hass, 'register_constraint')
 
     ## Initialize patches
     patched_run_daily = patch_run_daily.start()
@@ -45,6 +47,7 @@ def patch_hass():
     patched_call_service = patch_call_service.start()
     patched_turn_on = patch_turn_on.start()
     patched_turn_off = patch_turn_off.start()
+    patched_register_constraint = patch_register_constraint.start()
 
     ## Setup un-patch callback
     def unpatch_callback():
@@ -61,6 +64,7 @@ def patch_hass():
         patch_call_service.stop()
         patch_turn_off.stop()
         patch_turn_on.stop()
+        patch_register_constraint.stop()
 
     return ({
         'run_daily': patched_run_daily,
@@ -73,5 +77,6 @@ def patch_hass():
         'passed_args': patched_passed_args,
         'call_service': patched_call_service,
         'turn_on': patched_turn_on,
-        'turn_off': patched_turn_off
+        'turn_off': patched_turn_off,
+        'register_constraint': patched_register_constraint
     }, unpatch_callback)

